gratipay/models/participant/email.py

```diff
@@ -280,11 +281,19 @@ def save_email_address(self, cursor, address):
     def finish_package_claims(self, cursor, nonce, *packages):
         """Create teams if needed and associate them with the packages.
         """
-        cursor.run('DELETE FROM claims WHERE nonce=%s', (nonce,))
-        package_ids = []
+        if not packages:
+            raise NoPackages()
+
+        package_ids, teams, team_ids = [], [], []
         for package in packages:
-            package.ensure_team(cursor, self)
             package_ids.append(package.id)
+            team = package.get_or_create_linked_team(cursor, self)
+            teams.append(team)
+            team_ids.append(team.id)
+        review_url = self.app.project_review_repo.create_issue(*teams)
+
+        cursor.run('DELETE FROM claims WHERE nonce=%s', (nonce,))
+        cursor.run('UPDATE teams SET review_url=%s WHERE id=ANY(%s)', (review_url, team_ids,))
```

Comment:
Is this stuff happening in a transaction? Maybe there's a test that verifies that or something - I'll look as I scroll down.

I didn't see one. Maybe you know? I can't really tell.